### PR TITLE
Update and fix error handling for new api class

### PR DESCRIPTION
**Pull request by SergioEstevao**
Refs #5245 

Test 1:
 - Login to WP.com account
 - Go to blog list
 - Go to server and invalidate password and token for that user
 - Try to do a refresh and check if the correct error message/dialog shows up

Test 2:
 - Login with WP.com account
 - Reduce the maximum file size or disk quota for media for that user on the server to 1Mb
 - Try to upload a media file and see if a proper error message shows up.

Needs review: @koke

### CODE REVIEW

File: WordPress/Classes/Services/MediaService.m [hunk 1]
```diff
@@ -293,7 +271,7 @@ - (NSError *)translateMediaUploadError:(NSError *)error {
             } break;
         }
         userInfo[NSLocalizedDescriptionKey] = errorMessage;
-        newError = [[NSError alloc] initWithDomain:WordPressComApiErrorDomain code:errorCode userInfo:userInfo];
+        newError = [[NSError alloc] initWithDomain:WordPressComRestApiErrorDomain code:errorCode userInfo:userInfo];
```
Comment (koke) [inline WordPress/Classes/Services/MediaService.m]: I get that this was here before, but why is an _XML-RPC_ error on _MediaService_ converted into a `WordPressComRestApiErrorDomain`?

**Comment by koke** [comment]
I'm not really sure what this PR is fixing, can you add some context?

**Comment by SergioEstevao** [comment]
@koke The context for these issues are:
 -  WordPressComApiErrorDomain and WordPressComRestApiErrorDomain have different string values so the code in WPError wasn't catching these specific error anymore.
 - In the MediaService we have problem has above, and I also decided to take out these specific media upload error messages based on the httpstatus code that weren't reliable and just use the message that comes from the server.

**Comment by SergioEstevao** [comment]
@koke added some context, and merge the last changes on the 6.3 release branch, ready for another look.

File: WordPress/Classes/Utility/WPError.m [hunk 2]
```diff
@@ -82,9 +83,10 @@ + (void)showNetworkingAlertWithError:(NSError *)error title:(NSString *)title
             default:
                 break;
         }
-    } else if ([error.domain isEqualToString:WordPressComApiErrorDomain]) {
-        DDLogError(@"wp.com API error: %@: %@", [error.userInfo objectForKey:WordPressComApiErrorCodeKey], [error localizedDescription]);
-        if (error.code == WordPressComApiErrorInvalidToken || error.code == WordPressComApiErrorAuthorizationRequired) {
+    } else if ([error.domain isEqualToString:WordPressComRestApiErrorDomain]) {
```
Comment (koke) [inline WordPress/Classes/Utility/WPError.m]: Shouldn't we also keep the check for `WordPressComApiErrorDomain` until `WordPressComApi` is fully removed?

**Comment by koke** [comment]
Other than the comment about keeping error handling for `WordPressComApiErrorDomain` this looks fine :shipit:

**Pull request merged**